cxx/structureTranslationModule/translator/FromConceptTranslator.cpp

```diff
@@ -0,0 +1,39 @@
+#include "FromConceptTranslator.hpp"
+
+namespace structureTranslationModule
+{
+FromConceptTranslator::FromConceptTranslator(ScMemoryContext * context)
+  : StructureTranslator(context)
+{
+}
+
+std::vector<std::string> FromConceptTranslator::translate(ScAddr const & structAddr) const
+{
+  std::vector<std::string> translations;
+  std::string const NODE_ALIAS = "_node";
+  std::string const CLASS_ALIAS = "_class";
+  std::string const EDGE_ALIAS = "_edge";
+  ScAddr classNode, node;
+
+  ScTemplate scTemplate;
+  scTemplate.Triple(ScType::NodeVarClass >> CLASS_ALIAS, ScType::EdgeAccessVarPosPerm >> EDGE_ALIAS, ScType::NodeVar >> NODE_ALIAS);
+  scTemplate.Triple(structAddr, ScType::EdgeAccessVarPosPerm, EDGE_ALIAS);
+  ScTemplateSearchResult searchResult;
+  context->HelperSearchTemplate(scTemplate, searchResult);
```

Comment:
Checking only the edge is not correct (especially in other translators), you should check all the elements. To make it easy and effective you can use SmartTemplateSearch, just check all items with HelperCheckEdge from structure addr